TUM Campus App/LoginViewController.swift

```diff
@@ -58,39 +58,53 @@ extension LoginViewController: UITextFieldDelegate, TokenFetcherControllerDelega
     
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
         let replaced = NSString(string: textField.text ?? "").replacingCharacters(in: range, with: string)
-        switch textField {
-        case firstTextField:
-            if replaced.characters.count == 2 {
-                firstTextField.text = replaced
-                numbersTextField.becomeFirstResponder()
-                return false
-            }
-            break
-        case numbersTextField:
-            if replaced.characters.count == 2 {
-                numbersTextField.text = replaced
-                secondTextField.becomeFirstResponder()
-                return false
-            } else if replaced == "" {
-                numbersTextField.text = ""
-                firstTextField.becomeFirstResponder()
-                return false
-            }
-            break
-        case secondTextField:
-            if replaced.characters.count == 3 {
-                secondTextField.text = replaced
-                performSegue(withIdentifier: "waitForConfirmation", sender: self)
-                return false
-            } else if replaced == "" {
-                secondTextField.text = ""
-                numbersTextField.becomeFirstResponder()
-                return false
+        let numberOfCharacterPerTextField = [2, 2, 3]
+        let textFields = [firstTextField, numbersTextField, secondTextField]
+        
+        guard let textFieldIndex = textFields.index(where: { $0 == textField })
+            else { return true }
+        
+        if replaced == "" {
+            textField.text = ""
+            textFields[max(0, textFieldIndex - 1)]?.becomeFirstResponder()
+            return false
+        } else if string.characters.count < range.length {
+            return true
+        }
+    
+        var mutableReplaced = replaced
+        
+        for i in textFieldIndex..<textFields.endIndex {
+            let startIndex = mutableReplaced.startIndex
+            let endIndex = replaced.index(startIndex, offsetBy: min(mutableReplaced.characters.count, numberOfCharacterPerTextField[i]) - 1)
+            let range = ClosedRange(uncheckedBounds: (lower: startIndex, upper: endIndex))
```

Comment:
Uhhh! Unchecked Bounds, scary stuff... ;) Are you sure about this. I understand that you don't want the range to throw a fatalError if endIndex < startIndex.

Care to elaborate on why that happens?

Reply:
Hmm, what would be the alternative to using `uncheckedBounds`? The `endIndex < startIndex` would only ever be true if one of the `numberOfCharactersPerTextField` were to be `0` or less. Perhaps I should be explicit about that being an array of unsigned integers.

Reply:
`(startIndex..<endIndex)` should also work, but this is fine